v3_async_wip/v3_async_wip/iothub_session.py

```diff
@@ -0,0 +1,284 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+import asyncio
+import contextlib
+import ssl
+from typing import Optional, Union, AsyncGenerator, Type
+from types import TracebackType
+
+from v3_async_wip import signing_mechanism as sm
```

Comment:
so i don't always like so short names so something like sign_mech and conn_str but it is upto u